elf/icf.cc

```diff
@@ -106,12 +118,12 @@ static void uniquify_cies(Context<E> &ctx) {
 }
 
 template <typename E>
-static bool is_eligible(InputSection<E> &isec) {
+static bool is_eligible(Context<E> &ctx, InputSection<E> &isec) {
```

Comment:
For now I've adopted the LLD behavior where addr_sig sections does not participate in ICF at all -- though I wonder if it's legal to merge other non-addr_sig sections into one of the addr_sig section.

Reply:
Are you asking if it is a correct thing to consider all sections in an object file as if they were address-taken if the object file does not have a `.llvm_addrsig` section? If so, I think the answer is yes. That's a conservative assumption required for correctness.

Reply:
That’s not what I meant — sorry it wasn’t clear enough.

What I tried to say is to merge a section that doesn’t get its address taken into another section that has its address taken. It sounds safe because only the latter section can observe the “address”. But this merging currently doesn’t happen right now because address significant regions are completely removed from the ICF graph.

Reply:
I think we cannot merge two address-taken functions, but we can merge a non-address-taken function with an address-taken function. So the lld's behavior seems too conservative.

Reply:
I see. I'll try to measure if there's a significant difference. If there is, we could implement the more aggressive behavior.

Reply:
Thanks. If it will make a difference, please report it to lld (https://github.com/llvm/llvm-project/issues) with a reference to your patch so that they can fix the issue too.